### PR TITLE
feat: Add API Connector utilities

### PR DESCRIPTION
**Pull request by ondrosh**
## Changes

- Add `connectToAPI` method on `feathery.http` to utilize new API Connectors built by the no-code rule builder

### CODE REVIEW

File: src/utils/formHelperFunctions.ts [hunk 1]
```diff
@@ -820,5 +826,14 @@ export function httpHelpers(client: any) {
         headers: Record<string, string>
       ) => client.runCustomRequest(method, url, data, headers))
   );
+
+  helpers.connectToAPI = async (name: string) => {
```
Comment (bo-dun-1) [inline src/utils/formHelperFunctions.ts:830]: should this actually just be integrated into the runCustomRequest interface? maybe the url parameter can be polymorphic and also support `name`? This way we have just one interface and users can still pass in custom data and headers to adjust the api connector if they want

Comment (ondrosh) [reply src/utils/formHelperFunctions.ts:830]: Yeah, I can do that -- ideally we move completely away from the older payload all together in the future.

File: src/utils/client.ts [hunk 2]
```diff
@@ -530,20 +530,39 @@ export default class Client {
...
-      form_key: this.formKey,
-      method,
-      url,
-      user_data: userData,
-      headers
+      form_key: this.formKey
     };
+
+    if (typeof payload === 'string') {
+      Object.assign(data, {
```
Comment (bo-dun-1) [inline src/utils/client.ts:550]: data.name = payload;

File: src/utils/client.ts [hunk 3]
```diff
@@ -530,20 +530,39 @@ export default class Client {
...
+        name: payload
+      });
+    } else {
+      Object.assign(data, {
```
Comment (bo-dun-1) [inline src/utils/client.ts:554]: why not just assign one-by-one?

data.method = payload.method;
data.url = payload.url;

etc.

File: src/utils/formHelperFunctions.ts (hunk 1 above)
Comment (bo-dun-1) [reply src/utils/formHelperFunctions.ts:844]: i think we were going to rename this to connect as well

File: src/utils/formHelperFunctions.ts [hunk 4]
```diff
@@ -811,15 +817,45 @@ export function mapFormSettingsResponse(res: any) {
   };
 }
 
-export function httpHelpers(client: any) {
+export function httpHelpers(client: any, connectorFields: string[] = []) {
   const helpers: Record<string, any> = {};
   ['GET', 'PATCH', 'POST', 'PUT', 'DELETE'].forEach(
```
Comment (bo-dun-1) [inline src/utils/formHelperFunctions.ts:822]: could we also support lowercase get, patch, etc.?

**Review by bo-dun-1 — APPROVED** [review]

**Pull request merged**